src/shared/typed-fetch.ts

```diff
@@ -0,0 +1,9 @@
+export async function typedFetch<T>(url: URL): Promise<T> {
+  const data = await fetch(url);
+
+  if (!data.ok) {
+    throw new Error(`HTTP error! status: ${data.status}`);
```

Comment:
is there a pattern to pass the response body back, instead of text, pass an object with the error message like the one here and any error message that's sent from the server, so the client can display the server error message to the user

Reply:
Well one could of course create a custom Error class and use that. I think it would be nice for consumers of the components to have this extra information in the error. I've added it